compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/BIRPackageSymbolEnter.java

```diff
@@ -445,15 +448,43 @@ private void defineFunction(DataInputStream dataInStream) throws IOException {
                     for (int i = 0; i < resourcePathCount; i++) {
                         resourcePath.add(names.fromString(getStringCPEntryValue(dataInStream)));
                     }
+                    
+                    List<Location> resourcePathSegmentPosList = new ArrayList<>(resourcePathCount);
+                    for (int i = 0; i < resourcePathCount; i++) {
+                        resourcePathSegmentPosList.add(readPosition(dataInStream));
+                    }
 
                     Name accessor = names.fromString(getStringCPEntryValue(dataInStream));
                     BTupleType resourcePathType = (BTupleType) readBType(dataInStream);
                     
                     BResourceFunction resourceFunction = new BResourceFunction(names.fromString(funcName), 
                             invokableSymbol, funcType, resourcePath, accessor, pathParams, restPathParam,
                             resourcePathType, symTable.builtinPos);
-                    
-                    ((BStructureTypeSymbol) attachedType.tsymbol).attachedFuncs.add(resourceFunction);
+
+                    List<BType> resourcePathTypes = resourceFunction.resourcePathType.tupleTypes;
+                    BType restPathParamType = resourceFunction.resourcePathType.restType;
+                    if (restPathParamType != null) {
+                        resourcePathTypes.add(restPathParamType);
```

Comment:
does This modify resourceFunction.resourcePathType.tupleTypes?

Reply:
Yes, will update